workers/unity/Packages/io.improbable.gdk.tools/GenerateCode.cs

```diff
@@ -149,33 +150,76 @@ private static void Generate()
                 }
 
                 var workerJsonPath = Path.GetFullPath(Path.Combine(Application.dataPath, ".."));
+                var loggerOutputPath = Path.GetFullPath(Path.Combine(CodegenExeDirectory, "codegen-output.log"));
 
                 using (new ShowProgressBarScope("Generating code..."))
                 {
-                    var errorMessage = new StringBuilder();
+                    latestCodegenLogs = new List<CodegenLog>();
+
                     var exitCode = RedirectedProcess.Command(Common.DotNetBinary)
-                        .WithArgs(ConstructArgs(CodegenExe, schemaCompilerPath, workerJsonPath))
+                        .WithArgs(ConstructArgs(CodegenExe, schemaCompilerPath, workerJsonPath, loggerOutputPath))
                         .RedirectOutputOptions(OutputRedirectBehaviour.None)
-                        .AddErrorProcessing((line) => errorMessage.Append($"\n{line}"))
-                        .AddErrorProcessing(Debug.LogError)
-                        .AddOutputProcessing(ProcessStdOut)
+                        .AddOutputProcessing(ProcessDotnetOutput)
+                        .AddOutputProcessing(ProcessCodegenOutput)
                         .Run();
 
+                    var numWarnings = latestCodegenLogs.Count(line => line.Level == CodegenLogLevel.Warn);
+                    var numErrors = latestCodegenLogs.Count(line => line.Level == CodegenLogLevel.Error);
+
                     if (exitCode.ExitCode != 0)
                     {
                         if (!Application.isBatchMode)
                         {
-                            Debug.LogError($"Error(s) compiling schema files!{errorMessage}");
+                            Debug.LogError(File.Exists(loggerOutputPath)
+                                ? $"Code generation failed! Please check the code generation logs for more information: {loggerOutputPath}"
+                                : "Code generation failed! Please check the console for more information.");
+
```

Comment:
This information will be in the console regardless right?

Reply:
Also I like ternary expressions, but.. not this one 😛 

(Maybe refactor the string assignment out of the `Debug.LogError` call?